src/status_im/protocol/handlers.cljs

```diff
@@ -1,291 +1,204 @@
 (ns status-im.protocol.handlers
-  (:require [status-im.utils.handlers :as u]
-            [re-frame.core :refer [dispatch after] :as re-frame]
-            [status-im.utils.handlers :refer [register-handler]]
+  (:require [re-frame.core :as re-frame]
+            [status-im.utils.handlers :as handlers]
             [status-im.data-store.contacts :as contacts]
             [status-im.data-store.messages :as messages]
             [status-im.data-store.pending-messages :as pending-messages]
             [status-im.data-store.processed-messages :as processed-messages]
             [status-im.data-store.chats :as chats]
             [status-im.protocol.core :as protocol]
-            [status-im.constants :refer [text-content-type
-                                         blocks-per-hour]]
-            [status-im.i18n :refer [label]]
+            [status-im.constants :as constants]
+            [status-im.i18n :as i18n]
             [status-im.utils.random :as random]
             [status-im.protocol.message-cache :as cache]
-            [status-im.utils.datetime :as dt]
+            [status-im.utils.datetime :as datetime]
             [taoensso.timbre :as log :refer-macros [debug]]
-            [status-im.constants :as c]
             [status-im.native-module.core :as status]
-            [clojure.string :refer [join]]
-            [status-im.utils.scheduler :as s]
-            [status-im.utils.web3-provider :as w3]
-            [status-im.utils.datetime :as time]))
+            [clojure.string :as string]
+            [status-im.utils.web3-provider :as web3-provider]))
+
+;;;; COFX
+
+(re-frame/reg-cofx
+  ::get-web3
+  (fn [coeffects _]
+    (assoc coeffects :web3 (web3-provider/make-web3))))
+
+(re-frame/reg-cofx
+  ::get-chat-groups
+  (fn [coeffects _]
+    (assoc coeffects :groups (chats/get-active-group-chats))))
```

Comment:
Minor point, but @yenda already started moving relevant `co-effects`/`effects` into the `data-store` namespaces in his last PR and I think it's a right decision, as those should be easy to find, and hard to accidentally duplicate (as opposed to defining them in the `handlers`/`events` namespaces).

Reply:
yeah i'm thinking about this too, good point, but to be honest i don't want to delay this PR, so i'll prepare another one, because i'm not sure we need to use data-store here, waiting answer from @rasom , idk why we can't read this from app-db